iguana/xml_reader.hpp

```diff
@@ -80,6 +75,32 @@ inline void parse_attribute(rapidxml::xml_node<char> *node, T &t) {
       if (ec != std::errc{})
         throw std::invalid_argument("Failed to parse number");
       value_item = static_cast<value_type>(num);
+    } else if constexpr (is_std_variant_v<value_type>) {
```

Comment:
I think use std::visit will be better, and the code will be simple and clean.

```c++
struct visitor {
  template<typename T>
  void operator()(const T& t) {
    if constexpr (std::is_arithmetic_v<T>) {
      std::cout << typeid(t).name() << "\n";
    }
    else if constexpr (std::is_same_v<std::string, std::remove_cvref_t<T>>) {
      std::cout << typeid(t).name() << "\n";
    }
    else {
      static_assert(!sizeof(T), "don't support this type!");
    }
  }
};

void test_variant() {
  std::variant<int, float, double, std::string> v = 1;
  std::visit(visitor{}, v);

  std::variant<int, float, double, std::string> v1 = 2.5;
  std::visit(visitor{}, v1);

  std::variant<int, float, double, std::string> v2 = "test";
  std::visit(visitor{}, v2);
}
```

Reply:
use std::monostate as default.
std::variant<std::monostate, int, float, double, std::string>